Elastic format data

nkhadka21 opened the pull request:
This PR implements a routine to generate lightcurves in a given cadence.

sibirrer commented on data/SkyPy/lsst-like.yml:
```diff
@@ -34,7 +34,7 @@ tables:
       coefficients: $blue.coeff
       filter:  bessell-B
-    mag_g, mag_r, mag_i, mag_z, mag_Y: !skypy.galaxies.spectrum.kcorrect.apparent_magnitudes
+    mag_g, mag_r, mag_i, mag_z, mag_y: !skypy.galaxies.spectrum.kcorrect.apparent_magnitudes
```
is there a reason for the re-naming. Historically, the band Y is typed in capital letters. Not that I care much

nkhadka21 replied:
DP0 has convention of mag_y. So, I used this to make it consistent with DP0.

sibirrer commented on slsim/LsstSciencePipeline/util_lsst.py:
```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+def optimized_transient_event_time_mjd(
+    mjd_times, lightcurve_range=(-50, 100), min_points=100, optimized_cadence=True
+):
+    """Finds a transient start time such that at least min_points fall within
```
I don't really understand what min_points means and what this function exactly does from the documentation

nkhadka21 replied:
I removed this function!

sibirrer commented on slsim/LsstSciencePipeline/util_lsst.py:
```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+        interval where given minimum number of observation points occur.
+    :param min_points: minimum number of observations in a certain time
+        window.
+    :param noise: Boolean. If True, a gaussian noise is added to the
```
Gaussian noise in flux, I would guess. Can you change it to poisson? (then it's easier to get upper limits)

nkhadka21 replied:
It includes combination of background noise and poisson noise. This combined noise is added in a flux unit as a gaussian noise. Also, computes lower and upper asymmetric error in magnitude unit.

sibirrer commented on slsim/LsstSciencePipeline/util_lsst.py:
```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+        lightcurve flux.
+    :param symmetric: Boolean. If True, a symmetric error on magnitude
+        is provided.
+    :return: Astropy table of lightcurve and exposure information of dp0
```
the output format is not well described

sibirrer commented on slsim/LsstSciencePipeline/util_lsst.py:
```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+    :return: Astropy table of lightcurve and exposure information of dp0
+        data
+    """
+
```
this definition is very long and hard to read. Can you write it more compactly? Do you need to know the number of images (4)?

sibirrer commented on slsim/LsstSciencePipeline/util_lsst.py:
```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+    return copied_exposure_data
+
+
+def extract_lightcurves_in_different_bands(data, images=True):
```
is this definition (and the other one above) specific to LSST? Perhaps you can move this definition to another file

nkhadka21 replied:
This is not specific to the LSST but it demands the exposure data in the form of opsim simulation. So, It might be more clear to put this function here??

sibirrer commented on slsim/Plots/plot_functions.py:
```diff
@@ -129,3 +129,156 @@ def plot_montage_of_random_injected_lens(image_list, num, n_horizont=1, n_vertic
     return fig
+
+
+def plot_lightcurves(data, images=True):
```
it might be better to have more specific inputs in this definition to make it focused on the plotting

sibirrer reviewed (changes requested):
Thank you very much @nkhadka21! I have some comments and suggestions.

nkhadka21 commented:
> Just checking, did you address there comments such that it is for re-review?

Hi @sibirrer , I have not addressed these comments. I will address them soon.

sibirrer commented:
> Hi @sibirrer, I worked on this PR after long time. Romoved some of the unnecessary code and made it siomple so that It can be improved easily in the future. I have also included a notebook showing the use of this routine. Thank you!

Thank you @nkhadka21 ! I think this is good (enough) for a start. Thanks for finishing it. I think this might also help with @timedilatesme @padma18-vb (if it's not too late since they are already implementing something)
